pkg/generators/cluster/managedclusterworkercores.go

```diff
@@ -29,6 +29,11 @@ func GetManagedClusterWorkerCoresMetricFamilies(hubClusterID string) metric.Fami
 
 			core_worker, _ := getCapacity(mc)
 
+			// set worker cores to 0 if the ClusterDeployment is in hibernating state.
+			if isHibernatingFn(mc.GetName()) {
+				core_worker = 0
```

Comment:
From the metric, if a cluster's work core is 0, can we determine whether it is due to hibernation or another reason? should we log something here?

Reply:
Good point. I added the cluster state transition message  in `clusterHibernatingStateCache` instead: https://github.com/stolostron/clusterlifecycle-state-metrics/pull/452/files#diff-9a60ea0d41dcdb56fd35687cfa497bd121ca330e8b8a1f002e79728e2312455fR93-R97 #